### PR TITLE
Use PickerAvoidingView from react-native-picker-select

### PR DESCRIPTION
**Pull request by jakub-trzebiatowski**
### Details
Use PickerAvoidingView from react-native-picker-select to ensure pickers aren't covered by iOS picker modal.

Depends on https://github.com/Expensify/react-native-picker-select/pull/8

### Fixed Issues

$ https://github.com/Expensify/App/issues/15483
PROPOSAL: https://github.com/Expensify/App/issues/15483#issuecomment-1494587184

### Tests

On iOS:
1. On any screen, tap a picker that's on the bottom of the screen, colliding with the space modal needs
    - [ ] Ensure that the picker was "pushed" over the modal that appeared
2. Tap a picker that's on the top of the screen, not colliding with the space modal needs
    - [ ] Ensure that no glitches or unexpected behavior occur
3. Perform various picker interactions, like opening and closing them repeatedly
    - [ ] Ensure that no glitches or unexpected behavior occur
  
On other platforms:
1. Perform various picker interactions, like opening and closing them repeatedly
    - [ ] Verify that this PR did not change the behavior in any observable way

On all platforms:
1. Fill out a form, leaving only a required checkbox unselected
2. Press "Fix the errors"
    - [ ] Verify that the screen didn't scroll all the way up

### QA Steps

See [Tests](#tests)

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/2590174/231876624-42c9c0a1-1d54-47a6-a121-73e1b88c6513.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/2590174/231876518-eab5caca-4dcc-4c82-a516-1a2e64b61f79.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/2590174/231876388-5028adbc-2959-45f8-9bf0-b8748568f213.mp4

</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,7 +107,7 @@
     "react-native-pdf": "^6.6.2",
     "react-native-performance": "^4.0.0",
     "react-native-permissions": "^3.0.1",
-    "react-native-picker-select": "git+https://github.com/Expensify/react-native-picker-select.git#84ee97dec11c2e65609511eb5a757d61bbeeab79",
+    "react-native-picker-select": "git+https://github.com/Expensify/react-native-picker-select.git#683fd45e3ca18054e2d3ceb319e0cf595a1e26ed",
```
Comment (cubuspl42) [inline package.json:110]: We have to ensure this SHA1 is up-to-date at the end of the review.

Let's resolve this thread last.

File: src/components/ScreenWrapper/index.js [hunk 2]
```diff
@@ -110,19 +111,21 @@ class ScreenWrapper extends React.Component {
...
-                                            insets,
-                                            safeAreaPaddingBottomStyle,
-                                            didScreenTransitionEnd: this.state.didScreenTransitionEnd,
-                                        })
-                                        : this.props.children
-                                }
-                                {this.props.isSmallScreenWidth && (
-                                    <OfflineIndicator />
-                                )}
+                                <PickerAvoidingView>
```
Comment (cubuspl42) [inline src/components/ScreenWrapper/index.js:114]: From the user's perspective, things looks like this:

https://user-images.githubusercontent.com/2590174/231263997-5966da44-6ec6-4610-837e-14b1de13fcbc.mp4

I'd like to confirm that this is the expected effect.

**Comment by cubuspl42** [comment]
@0xmiroslav I'm still finishing the author's checklist, but I would say that both PRs are ready for review! It would be _awesome_ if we managed to wrap this in three days, and I think that's realistic 😊

File: src/components/ScreenWrapper/index.js (hunk 2 above)
Comment (cubuspl42) [reply src/components/ScreenWrapper/index.js:114]: @0xmiroslav Was this meant to be video? Because it's PNG, I'm not sure how that happened

Comment (cubuspl42) [reply src/components/ScreenWrapper/index.js:114]: I'm sorry, I confused the behavior I remembered from iOS Safari / Web. I have a solution that seems acceptable. I'll push it in a few moments.

File: package.json (hunk 1 above)
Comment (cubuspl42) [reply package.json]: I used the path to the `Expensify` fork, but that might've been a bad idea. Fixed.

File: src/components/ScreenWrapper/index.js (hunk 2 above)
Comment (cubuspl42) [reply src/components/ScreenWrapper/index.js]: @0xmiroslav @Julesssss 

This is how the solution looks with a subtle animation:

https://user-images.githubusercontent.com/2590174/231567854-2f5f7de5-6c5a-460c-94bc-0bdef46bf182.mp4

File: package.json [hunk 3]
```diff
@@ -107,7 +107,7 @@
     "react-native-pdf": "^6.6.2",
     "react-native-performance": "^4.0.0",
     "react-native-permissions": "^3.0.1",
-    "react-native-picker-select": "git+https://github.com/Expensify/react-native-picker-select.git#84ee97dec11c2e65609511eb5a757d61bbeeab79",
+    "react-native-picker-select": "git+https://github.com/cubuspl42/react-native-picker-select.git#039abcb9cd6e444538537e3da5c1af1ad970ab8d",
```
Comment (cubuspl42) [inline package.json:110]: At the end of the PR review:
* Verify that the SHA1 is up-to-date
* **Replace the prefix** with `Expansify`

Let's keep this thread open.

**Review by Julesssss — COMMENTED** [review]
Looking good, but we still need to run and test on other platforms

**Comment by cubuspl42** [comment]
> @cubuspl42 please pull from main

Ok! I'll get back to this issue in 2 hours or so, and then I'll pull from main and finish the checklist.

File: src/components/ScreenWrapper/index.js (hunk 2 above)
Comment (cubuspl42) [reply src/components/ScreenWrapper/index.js]: I understand the problem now. I'm on it.

Comment (cubuspl42) [reply src/components/ScreenWrapper/index.js]: Disabled:

https://user-images.githubusercontent.com/2590174/231866214-775d03d8-8c53-436e-bdb2-aba168b138b9.mp4

Enabled:

https://user-images.githubusercontent.com/2590174/231866227-f69df8e9-928d-437d-908d-00e13709eb8d.mp4

File: package.json (hunk 3 above)
Comment (cubuspl42) [reply package.json]: At the moment of writing, in the newest revision, the hash is up-to-date. We'll still need to switch to the `Expensify` prefix once the other PR is merged.

**Review by 0xmiroslav — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (cubuspl42) [reply package.json]: Up-to date again, prefix still to be updated

**Review by 0xmiroslav — DISMISSED** [review]

**Review by Julesssss — COMMENTED** [review]
Thanks, looking good. Just left a comment on the library PR.

File: package.json (hunk 3 above)
Comment (cubuspl42) [reply package.json]: SHA1 & prefix updated

**Comment by cubuspl42** [comment]
@0xmiroslav Of course... Done.

**Review by 0xmiroslav — DISMISSED** [review]
iOS tests well. Confirmed other platforms are not affected.

3 different console errors on iOS, already happens on main so out of scope:

https://user-images.githubusercontent.com/97473779/234562986-762f523d-90d0-4d73-a5e9-57ca7205a155.mov

cc: @Julesssss

File: src/pages/AddPersonalBankAccountPage.js [hunk 4]
```diff
@@ -85,7 +85,7 @@ class AddPersonalBankAccountPage extends React.Component {
         const shouldShowSuccess = lodashGet(this.props, 'personalBankAccount.shouldShowSuccess', false);
 
         return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={shouldShowSuccess}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={shouldShowSuccess} shouldEnablePickerAvoiding={false}>
```
Comment (Julesssss) [inline src/pages/AddPersonalBankAccountPage.js:88]: Sorry to be a pain, but would you mind putting the props on newlines now that there are two? Thanks!

```
<ScreenWrapper
    includeSafeAreaPaddingBottom={shouldShowSuccess}
    shouldEnablePickerAvoiding={false}
>
```

File: src/pages/ReimbursementAccount/BankAccountPlaidStep.js [hunk 5]
```diff
@@ -68,7 +68,7 @@ class BankAccountPlaidStep extends React.Component {
         const selectedPlaidAccountID = lodashGet(this.props.reimbursementAccountDraft, 'plaidAccountID', '');
 
         return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false} shouldEnablePickerAvoiding={false}>
```
Comment (Julesssss) [inline src/pages/ReimbursementAccount/BankAccountPlaidStep.js:71]: Same here please

File: src/pages/workspace/WorkspaceNewRoomPage.js [hunk 6]
```diff
@@ -138,7 +138,7 @@ class WorkspaceNewRoomPage extends React.Component {
         }));
 
         return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false} shouldEnablePickerAvoiding={false}>
```
Comment (Julesssss) [inline src/pages/workspace/WorkspaceNewRoomPage.js:141]: And here

File: src/pages/workspace/WorkspacePageWithSections.js [hunk 7]
```diff
@@ -106,7 +106,7 @@ class WorkspacePageWithSections extends React.Component {
         const policyName = lodashGet(this.props.policy, 'name');
 
         return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false} shouldEnablePickerAvoiding={false}>
```
Comment (Julesssss) [inline src/pages/workspace/WorkspacePageWithSections.js:109]: And here

**Review by Julesssss — COMMENTED** [review]
Looking good, just requested some minor formatting changes.

File: src/pages/AddPersonalBankAccountPage.js (hunk 4 above)
Comment (cubuspl42) [reply src/pages/AddPersonalBankAccountPage.js:88]: Done! (& merged main)

**Review by Julesssss — COMMENTED** [review]
Thanks, just need @0xmiroslav to review again and we're good 👍

**Review by 0xmiroslav — APPROVED** [review]
Tested again. Nothing breaks

**Review by Julesssss — APPROVED** [review]

**Pull request merged**